src/persistent_log/mem.rs

```diff
@@ -86,18 +86,18 @@ impl Log for MemLog {
         }
     }
 
-    fn entry(&self, index: LogIndex) -> result::Result<(Term, &[u8]), Error> {
+    fn entry(&self, index: LogIndex) -> result::Result<(Term, Vec<u8>), Error> {
         let (term, ref bytes) = self.entries[(index - 1).as_u64() as usize];
-        Ok((term, &bytes))
+        Ok((term, bytes.to_owned()))
     }
 
     fn append_entries(&mut self,
                       from: LogIndex,
-                      entries: &[(Term, &[u8])])
+                      entries: &[(Term, Vec<u8>)])
```

Comment:
I think that part of the change is misguided (why give something owned (the `Vec`) to something that is likely going to "copy" in some way anways?).
Building the `Vec<(Term, &[u8])>` in the callers of this method just for the one call feels kind of weird too, though. Could we get away with a signature like below?

```rust
fn append_entries(&mut self, from: LogIndex, term: Term, entries: &&[u8])
```

That might mean having to call `append_entries` multiple times if there are entries for multiple terms, but that's unlikely to happen and, when it does, acceptable.